src/network/connection.cpp

```diff
@@ -1173,7 +1173,9 @@ Connection::Connection(u32 protocol_id, u32 max_packet_size, float timeout,
 	m_bc_peerhandler(peerhandler)
 
 {
-	m_udpSocket.setTimeoutMs(5);
+	/* Amount of time Receive() will wait for data, this is entirely different
+	 * from the connection timeout */
+	m_udpSocket.setTimeoutMs(500);
```

Comment:
that can cause the server to hang i think. We may distinct the client acceptable version and the server. I think 10 ms is acceptacle on server but not 500, this will hang the entire network layer

Reply:
This runs on the `ConnectionReceiveThread` which has no job to do other than receiving.